build/openwisp_base/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM python:3.7-alpine AS BASE
+
+COPY ./openwisp_base/requirements.txt /opt/openwisp/requirements.txt
+RUN apk add --update --no-cache \
+    --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing \
+    gdal-dev geos-dev zlib-dev jpeg-dev openssl-dev libffi-dev gettext gcc && \
+    apk add --update --virtual .build-deps postgresql-dev git build-base linux-headers && \
+    pip install --no-cache-dir -r /opt/openwisp/requirements.txt && \
```

Comment:
I see now that you do everything in one step, are you doing it to save on image size?

I would separate the part which deal with system packages and the part which deals with the python packages

Reply:
It's done in one step because `.build-deps` are installed and uninstalled in the same layer to avoid taking useless files to next layer. It's a common practice I've noticed in a lot of other images as well. 
The system packages not in `.build-deps` don't take a lot of time to install, the real thing that takes a lot of time is `pip install` only, hence, I coupled all of them in one layer.

Should I keep `.build-deps` with python packages only or move them seperately as well?
(That makes dockerfiles look cleaner but bloats the image to about twice the size.)

Reply:
how much size increase are we talking about? If it's less than 100 MB, proceed to separate them.

Reply:
After the new changes, the current size is ~450MBs and seperating will make it around ~600MBs.
(However, since this is the base image, all the derived image size will be increased.)